planner/core/testdata/plan_suite_out.json

```diff
@@ -1588,7 +1588,7 @@
     "Cases": [
       {
         "SQL": "select t1.a, (select count(t2.a) from t t2 where t2.g in (select t3.d from t t3 where t3.c = t1.a)) as agg_col from t t1;",
-        "Best": "Apply{IndexReader(Index(t.f)[[NULL,+inf]])->IndexHashJoin{IndexReader(Index(t.c_d_e)[[NULL,+inf]]->HashAgg)->HashAgg->IndexReader(Index(t.g)[[NULL,NULL]])}(test.t.d,test.t.g)}->HashAgg"
```

Comment:
Expected, IndexJoin and IndexHashJoin have no difference in model2, and whether to push the Agg down will be solved by https://github.com/pingcap/tidb/issues/38874